src/hooks/useResize/index.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from "react";
+
+/**
+ * `useResize` determines if the window is currently being resized or not.
+ * @returns a boolean indicating whether the window is being resized or not
+ */
+export const useResize = () => {
```

Comment:
To me this hook has an obvious use case of calling a function on resize—in this case we just only happen to care about the `isResizing` status. If you wanted I think it'd be handy to flesh this out with an `onResize` prop that gets called in the hook's `onResize` if it's provided. But that's definitely overkill so you could totally just leave a comment saying something to that effect if you like!